packages/react/src/Image.doc.ts

```diff
@@ -0,0 +1,44 @@
+import {ReferenceEntityTemplateSchema} from '@shopify/generate-docs';
+
+const data: ReferenceEntityTemplateSchema = {
+  name: 'Image',
+  category: 'components',
+  isVisualComponent: true,
+  related: [
+    {
+      name: 'MediaFile',
+      type: 'component',
+      url: 'api/hydrogen/components/product-variant/mediafile',
+    },
+  ],
+  description:
+    "The `Image` component renders an image for the Storefront API's\n[Image object](https://shopify.dev/api/storefront/reference/common-objects/image) by using the `data` prop. You can [customize this component](https://shopify.dev/api/hydrogen/components#customizing-hydrogen-components) using passthrough props.\n\nAn image's width and height are determined using the following priority list:\n1. The width and height values for the `loaderOptions` prop\n2. The width and height values for bare props\n3. The width and height values for the `data` prop\n\nIf only one of `width` or `height` are defined, then the other will attempt to be calculated based on the image's aspect ratio,\nprovided that both `data.width` and `data.height` are available. If `data.width` and `data.height` aren't available, then the aspect ratio cannot be determined and the missing value will remain as `null`",
```

Comment:
I wonder if this inline markdown text will get unwieldy?

Reply:
The feedback I left was that this should just pull from the definition in the TS file, instead of needing to manually copy this over by hand. So hopefully in the future it's better.